In [1]:
import numpy as np
import pandas as pd

## a. 99% Confidence Interval using Sample Standard Deviation

#### mean

In [2]:
import numpy as np

data = [1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48,
        1.20, 1.33, 1.18, 1.22, 1.29]

mean = np.mean(data)

mean

np.float64(1.2386666666666666)

###### The sample mean represents the average durability of the print-heads.
###### It is calculated by summing all observations and dividing by the sample size (n = 15).
###### This value serves as a point estimate of the population mean.

#### sample std

In [3]:
std = np.std(data, ddof=1)

std

np.float64(0.19316412956959936)

###### The sample standard deviation measures the variability in the durability of print-heads.
###### Since the data is a sample, we use sample standard deviation (with n−1 in denominator).
###### This is important for constructing the confidence interval using the t-distribution.

#### t-value

In [4]:
from scipy import stats

n = 15
alpha = 0.01

t_value = stats.t.ppf(1 - alpha/2, df=n-1)

t_value

np.float64(2.976842734370834)

###### Since the sample size is small (n < 30) and population standard deviation is unknown, the t-distribution is used.
###### For a 99% confidence level and degrees of freedom (df = 14), the corresponding t-value is obtained.
###### This t-value is used to construct the confidence interval.

#### margin of error

In [6]:
ME = t_value * (std / (n ** 0.5))

ME

np.float64(0.14846932822817596)

###### Margin of Error represents the maximum expected difference between the sample mean and the population mean.
###### It depends on the t-value, sample standard deviation, and sample size.
###### A higher confidence level results in a larger margin of error.

#### confidence interval

In [10]:
lower = mean - ME
upper = mean + ME

(lower, upper)

(np.float64(1.0901973384384906), np.float64(1.3871359948948425))

###### The 99% confidence interval for the mean durability is:
###### lower = 1.0901973384384906
###### upper = 1.3871359948948425

## b. 99% Confidence Interval (Population Std Known)

#### z-value

In [11]:
from scipy import stats

z_value = stats.norm.ppf(1 - 0.01/2)

z_value

np.float64(2.5758293035489004)

###### Since the population standard deviation is known, the normal (z) distribution is used.
###### For a 99% confidence level, the corresponding z-value is obtained.
###### This z-value is used to calculate the confidence interval.

#### Calculate the Margin of Error (ME)

In [13]:
sigma = 0.2
n = 15

ME_b = z_value * (sigma / (n ** 0.5))

ME_b

np.float64(0.13301525327090588)

###### Margin of Error is calculated using the z-value and known population standard deviation.
###### Since σ is known, it provides a more accurate estimate compared to sample-based estimation.
###### The margin of error determines the range around the sample mean.

#### Confidence Interval (CI)

In [14]:
lower_b = mean - ME_b
upper_b = mean + ME_b

(lower_b, upper_b)

(np.float64(1.1056514133957607), np.float64(1.3716819199375725))

###### 99% CI (using population std = 0.2)
###### lower_b = 1.1056514133957607
###### upper_b = 1.3716819199375725

## Conclusion

###### The confidence intervals were constructed using both t-distribution and z-distribution methods.
###### The interval using known population standard deviation is slightly narrower, indicating more precision.
###### These intervals provide an estimate of the true mean durability of print-heads with 99% confidence.